packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -4113,6 +4173,34 @@ export class Quaternion {
         }
     }
 
+    /**
+     * Creates the rotation quaternion needed to rotate from one Vector3 onto another Vector3
+     * @param fromVector the starting vector
+     * @param toVector the ending vector
+     * @returns the rotation quaternion needed
+     */
+    public static RotationQuaternionFromOnto(fromVector: DeepImmutable<Vector3>, toVector: DeepImmutable<Vector3>) {
+        const ref: Quaternion = Quaternion.Zero();
+        return Quaternion.RotationQuaternionFromOntoToRef(fromVector, toVector, ref);
+    }
+
+    /**
+     * Creates the rotation quaternion needed to rotate from one Vector3 onto another Vector3 and stores in a result Quaternion
+     * @param fromVector the starting vector
+     * @param toVector the ending vector
+     * @param result the rotation quaternion needed
+     * @returns the result
+     */
+    public static RotationQuaternionFromOntoToRef(fromVector: DeepImmutable<Vector3>, toVector: DeepImmutable<Vector3>, result: Quaternion) {
+        const normal: Vector3 = Vector3.Cross(fromVector, toVector);
+        if (normal.equals(Vector3.Zero())) {

```

Comment:
Better to use `Vector3.ZeroReadOnly` to avoid allocating a new vector

Reply:
Makes sense. Done. 
Also made more use of ToRef methods